relays/substrate-client/src/chain.rs

```diff
@@ -0,0 +1,87 @@
+// Copyright 2019-2020 Parity Technologies (UK) Ltd.
+// This file is part of Parity Bridges Common.
+
+// Parity Bridges Common is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity Bridges Common is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity Bridges Common.  If not, see <http://www.gnu.org/licenses/>.
+
+use crate::client::Client;
+
+use frame_support::Parameter;
+use jsonrpsee::common::{DeserializeOwned, Serialize};
+use sp_core::Pair;
+use sp_runtime::traits::{
+	AtLeast32Bit, AtLeast32BitUnsigned, Bounded, CheckEqual, Dispatchable, Header as HeaderT, MaybeDisplay,
+	MaybeMallocSizeOf, MaybeSerialize, MaybeSerializeDeserialize, Member, SimpleBitOps,
+};
+use sp_std::fmt::Debug;
+
+/// Substrate-based chain from minimal relay-client point of view.
+pub trait Chain {
+	/// The block number type used by the runtime.
+	type BlockNumber: Parameter
+		+ Member
+		+ MaybeSerializeDeserialize
+		+ Debug
```

Comment:
It seems that some of these bounds are already part of `Parameter` or `Member`. Do we really need them explicitly?

Reply:
It's a plain copy of `frame_system::Trait`. Actually we don't need all of these trait, but will probably need some of these later.